APCAppCore/APCAppCore/Library/DataArchiverAndUploader/APCDataArchiverAndUploader.m

```diff
@@ -1842,24 +1886,22 @@ - (void) finalCleanupHandlingError: (NSError *) error
     {
         APCLogError2 (error);
     }
-
-
-    /*
-     IMPORTANT!
-     #warning Ron - To Do:  handle "we couldn't upload to Sage" errors.  (How?)
-
-     Perhaps:
-
-     if (error == couldn't upload to Sage)
-     {
-        queue some try-again concept
-     }
-     else
-     {
-        truly clean up
-     }
-     */
-
+    
+    __weak typeof(self) weakSelf = self;
+    
+    if (self.uploaderCompletionCallback)
+    {
+        if (self.callbackQueue)
+        {
+            [self.callbackQueue addOperationWithBlock:^{
+                weakSelf.uploaderCompletionCallback(error);
+            }];
+        }
```

Comment:
Why not just do `[self.callbackQueue addOperationWithBlock: weakSelf.uploaderCompletionCallback]`?